Velocity analysis

matt-gust opened the pull request:
Added functions for velocity analysis as well as a vignette showing how these functions can be used in conjuction with scVelo in python

matt-gust committed:
Create placeholder.txt

added placeholder

matt-gust committed:
brought make_loom_files and make_environment into repo

matt-gust committed:
trying different things but still getting error code 127

matt-gust committed:
added echo statements to tell me what step I'm at

matt-gust committed:
removed ml velocyto

matt-gust committed:
removed "echo" statements that i added during troubleshooting

matt-gust committed:
removed placeholder.txt

matt-gust committed:
copied function over from rrr_scu_tmp and changed paths to make_loom_files and the yml

matt-gust committed:
moved these two files to inst

matt-gust committed:
added roxygen headers for r_make_loom_files and use_sbatch_template

matt-gust committed:
added system calls to make folders for output

matt-gust committed:
added sampleid to velocyto call and use_sbatch_template so I don't get weird tags on loom file names

matt-gust committed:
made tmp folders outside of loop and redid naming of tmp bcs

matt-gust committed:
added use of metadata parameter to velocyto run

matt-gust committed:
fixed typo

matt-gust committed:
changes paths so that I could test pipeline

matt-gust committed:
made slurmOut the appropriate folder

matt-gust committed:
removed metadata argument from velocyto

matt-gust committed:
removed use of metadata in r_make_loom_files

matt-gust committed:
added write_off_md to r functions

matt-gust committed:
added quarto to engines for generating vignettes

matt-gust committed:
created vignette for rna velocity analysis. up through making loom files

matt-gust committed:
changed write_off_md to include all reductions

matt-gust committed:
added a check for whether or not user provided directory ends with a / or not

matt-gust committed:
finished vignette

matt-gust committed:
fixed how I save off reductions

matt-gust committed:
finished vignette!

matt-gust committed:
documented new functions

matt-gust committed:
Merge branch 'for-testing' into velocity-analysis

matt-gust committed:
ran document on package

MVesuviusC commented on inst/make_loom_files.sh:
```diff
@@ -0,0 +1,45 @@
+then
+    echo "env already exists";
+else
+    conda create -p ${env_path} -f make_environment.yml
```
I think you may need to specify the full path to the make_environment.yml file. This likely works when you test it, but when someone else installs the package that file won't be in whatever folder they're working from. You can get the location of the file similar to I do here: https://github.com/kidcancerlab/rrrSnvs/blob/cf535ec028815b5edacf91ca7c985830e3cfff53/R/snp_call.R#L208

matt-gust committed:
removed creating environment statement since I'm already doing it in R

matt-gust committed:
fixed paths to make_environment.yml

matt-gust committed:
re-ran document()

matt-gust commented:
I removed the lines in make_loom_files.sh that optionally created the conda environment since I'm already making it with a system() call in the R function.

I also fixed the paths to make_environment.yml and make_loom_files.sh in the r_make_loom_files function to reflect them being dumped a folder up from inst

MVesuviusC commented on inst/make_loom_files.sh:
```diff
@@ -0,0 +1,37 @@
+#SBATCH --output=placeholder_slurm_out
+#SBATCH --error=placeholder_slurm_out
+#SBATCH --job-name rrr_make_loom_files
+#SBATCH --array=0-0
```
If you're not doing array jobs you don't need this line

MVesuviusC commented on inst/make_loom_files.sh:
```diff
@@ -0,0 +1,37 @@
+#SBATCH --nodes=1
+#SBATCH --ntasks=1
+#SBATCH --cpus-per-task=4
+#SBATCH --mem=80G
```
Check your slurm output to see how much ram you're actually using, double it and round up to a round number to figure out how much ram to use. Also check if you need four cpus

matt-gust replied:
Max RSS was ~38000M for each job, doubling and rounding up comes out to 80G actually ¯\_(ツ)_/¯

MVesuviusC commented on R/velocity_analysis_functions.R:
```diff
@@ -0,0 +1,276 @@
+#' Wrapper to make loom files in R
```
The title here should be Title Case

matt-gust committed:
installing scvelo from source

matt-gust committed:
made make_loom_files.sh work for array jobs

matt-gust committed:
reworked r_make_loom_files so that it submits an array job

MVesuviusC committed:
Making function titles Title Case

MVesuviusC committed:
Minor edits to the vignette

matt-gust committed:
made symlink to grab bam files instead of copying them

matt-gust committed:
added echo statement to make sure samples are right

matt-gust committed:
removed echo

matt-gust committed:
removed attempt at symlink

matt-gust committed:
Merge branch 'parallelizing' into velocity-analysis

matt-gust committed:
Merge remote-tracking branch 'origin/velocity-analysis' into velocity-analysis

matt-gust committed:
fixed typo

matt-gust committed:
ran devtools::document on latest changes

matt-gust commented:
I re-worked r_make_loom_files and make_loom_files.sh to be an array job, that way the loom files don't have to be created sequentially

matt-gust committed:
attempting to create slurm output with sample id appended to end

matt-gust committed:
changed output file to be named by sample id rather than slurm job id

matt-gust committed:
created loom directory before calling velocyto. array job throws error for one sample otherwise

matt-gust committed:
limited number of simultaneous array jobs to 10

matt-gust committed:
forcibly delete tmp bams folder

matt-gust committed:
making conda environment from appropriate channels now

matt-gust committed:
added some checks to output to make sure it works

matt-gust committed:
made it so that separate output is created for each object

matt-gust committed:
missing comma

matt-gust committed:
handled case where num cells in an object == 1

matt-gust committed:
fixed mixup with getting cell file from array. now i get it from the sample id

matt-gust committed:
Update velocity_analysis_functions.R

MVesuviusC committed:
Minor style edits

MVesuviusC committed:
Merge branch 'velocity-analysis' of https://github.com/kidcancerlab/rrrSingleCellUtils into velocity-analysis

MVesuviusC committed:
Added eval:FALSE for velocity analysis vignette

MVesuviusC reviewed (approved):
All good, yo

MVesuviusC reviewed (approved):
I think we're good

MVesuviusC committed:
Merge branch 'master' into velocity-analysis
